outport/notifier/eventNotifier.go

```diff
@@ -0,0 +1,210 @@
+
+	events := en.getLogEventsFromTransactionsPool(args.TransactionsPool.Logs)
+	log.Debug("extracted events from block logs", "num events", len(events))
+
+	blockData := SaveBlockData{
+		Hash:      hex.EncodeToString(args.HeaderHash),
+		Txs:       args.TransactionsPool.Txs,
+		Scrs:      args.TransactionsPool.Scrs,
+		LogEvents: events,
+	}
+
+	err := en.httpClient.Post(pushEventEndpoint, blockData, nil)
+	if err != nil {
+		return fmt.Errorf("%w in eventNotifier.SaveBlock while posting block data", err)
+	}
+
+	return nil
+}
+
+func (en *eventNotifier) getLogEventsFromTransactionsPool(logs []*nodeData.LogData) []Event {
+	var logEvents []nodeData.EventHandler
+	for _, logData := range logs {
+		if logData == nil {
+			continue
+		}
+		if check.IfNil(logData.LogHandler) {
+			continue
+		}
+
+		logEvents = append(logEvents, logData.LogHandler.GetLogEvents()...)
+	}
+
+	if len(logEvents) == 0 {
+		return nil
+	}
+
+	var events []Event
+	for _, eventHandler := range logEvents {
+		if !eventHandler.IsInterfaceNil() {
+			bech32Address := en.pubKeyConverter.Encode(eventHandler.GetAddress())
+			eventIdentifier := string(eventHandler.GetIdentifier())
+
+			log.Debug("received event from address",
+				"address", bech32Address,
+				"identifier", eventIdentifier,
+			)
+
+			events = append(events, Event{
+				Address:    bech32Address,
+				Identifier: eventIdentifier,
+				Topics:     eventHandler.GetTopics(),
+				Data:       eventHandler.GetData(),
+			})
+		}
+	}
+
+	return events
+}
+
+// RevertIndexedBlock converts revert data in order to be pushed to subscribers
+func (en *eventNotifier) RevertIndexedBlock(header nodeData.HeaderHandler, _ nodeData.BodyHandler) error {
+	blockHash, err := core.CalculateHash(en.marshalizer, en.hasher, header)
+	if err != nil {
+		return fmt.Errorf("%w in eventNotifier.RevertIndexedBlock while computing the block hash", err)
+	}
+
+	revertBlock := RevertBlock{
+		Hash:  hex.EncodeToString(blockHash),
+		Nonce: header.GetNonce(),
+		Round: header.GetRound(),
+		Epoch: header.GetEpoch(),
+	}
+
+	err = en.httpClient.Post(revertEventsEndpoint, revertBlock, nil)
+	if err != nil {
+		return fmt.Errorf("%w in eventNotifier.RevertIndexedBlock while posting event data", err)
+	}
+
+	return nil
+}
+
+// FinalizedBlock converts finalized block data in order to push it to subscribers
+func (en *eventNotifier) FinalizedBlock(headerHash []byte) error {
+	finalizedBlock := FinalizedBlock{
+		Hash: hex.EncodeToString(headerHash),
+	}
+
+	err := en.httpClient.Post(finalizedEventsEndpoint, finalizedBlock, nil)
+	if err != nil {
+		return fmt.Errorf("%w in eventNotifier.FinalizedBlock while posting event data", err)
+	}
+
+	return nil
+}
+
+// SaveRoundsInfo returns nil
+func (en *eventNotifier) SaveRoundsInfo(_ []*indexer.RoundInfo) error {
+	return nil
+}
+
+// SaveValidatorsRating returns nil
+func (en *eventNotifier) SaveValidatorsRating(_ string, _ []*indexer.ValidatorRatingInfo) error {
+	return nil
```

Comment:
might add tests for all these "mock" functions as to be sure the code does not panic when called

Reply:
added more tests